### ISSUE TITLE
Updating a running workspace devfile should not require to stop,restart and wait.

### ISSUE THREAD

**Issue by sunix** [eclipse-che/che#18431]
### Describe the bug
When the user is updating the devfile of a running workspace, most of the time he has to stop, restart the workspace and wait for it to be ready. He cannot continue to code while the new version is available.

### Expected behavior
When updating the devfile, a new pod should be created: the user should still use the previous pod until the new one is ready. Once it is ready, theia could then automatically reload and display the new workspace pod.

Nice to have, in Theia status bar, could indicate that the workspace is being restarted.

**Comment by ibuziuk** [comment]
putting P2 for now, but feel free to update priority based on the team availability
